internal/provider/license_resource.go

```diff
@@ -0,0 +1,194 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/coder/coder/v2/codersdk"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/int32planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ resource.Resource = &LicenseResource{}
+
+func NewLicenseResource() resource.Resource {
+	return &LicenseResource{}
+}
+
+// LicenseResource defines the resource implementation.
+type LicenseResource struct {
+	data *CoderdProviderData
+}
+
+// LicenseResourceModel describes the resource data model.
+type LicenseResourceModel struct {
+	ID        types.Int32  `tfsdk:"id"`
+	ExpiresAt types.Int64  `tfsdk:"expires_at"`
+	License   types.String `tfsdk:"license"`
+}
+
+func (r *LicenseResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_license"
+}
+
+func (r *LicenseResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "A license for a Coder deployment.\n\nTerraform does not guarantee this resource" +
```

Comment:
You should recommend using multiple license resources if you're trying to swap out a license, as terraform will delete the old one THEN create the new one when replacing which will result in a brief unlicensed period